src/bun_js.zig

```diff
@@ -418,7 +418,7 @@ pub const Run = struct {
 
         vm.onUnhandledRejection = &onUnhandledRejectionBeforeClose;
         vm.global.handleRejectedPromises();
-        if (this.any_unhandled and this.vm.exit_handler.exit_code == 0) {
+        if (this.any_unhandled and this.vm.exit_handler.exit_code == null) {
```

Comment:
This is a bug. I think it's better to revert it to the previous behavior.

Exit code 0:
```js
❯ bun-11492 /tmp/a.js
1 | process.exitCode = 0;
2 | queueMicrotask(() => {
3 |   throw new Error('oopsie');
            ^
error: oopsie
      at /private/tmp/a.js:3:9
```

Exit code 1:

```js
❯ bun /tmp/a.js
1 | process.exitCode = 0;
2 | queueMicrotask(() => {
3 |   throw new Error('oopsie');
            ^
error: oopsie
      at /private/tmp/a.js:3:9

Bun v1.1.21-canary.1+79ddf0e47 (macOS arm64)
```

Exit code 1:

```js
❯ node /tmp/a.js
/private/tmp/a.js:3
  throw new Error('oopsie');
        ^

Error: oopsie
    at /private/tmp/a.js:3:9
    at node:internal/process/task_queues:140:7
    at AsyncResource.runInAsyncScope (node:async_hooks:206:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)

Node.js v22.4.1
```